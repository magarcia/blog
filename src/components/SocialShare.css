@media only screen and (min-device-width: 480px) {
  .aside {
    position: fixed;
    top: 200px;
    visibility: hidden;
    opacity: 0;
    transition: visibility 0s linear 0.3s, opacity 0.3s 0s;
  }
  .aside.visible {
    visibility: visible;
    opacity: 1;
    transition: visibility 0s linear 0s, opacity 0.3s 0s;
  }
}
.aside a:not(.anchor):not(.gatsby-resp-image-link) {
  font-size: 17px;
  color: var(--text);
  opacity: 0.65;
  position: relative;
  background: var(--bg);
  text-decoration: none;
  cursor: pointer;
  user-select: none;
  border: none;
  box-sizing: border-box;
  margin: 10px 3px;
  height: 24px;
  background: transparent !important;
}
@media only screen and (min-device-width: 480px) {
  .aside a:not(.anchor):not(.gatsby-resp-image-link) {
    display: block;
  }
}
.aside a:hover {
  opacity: 1;
}
.aside a:focus {
  outline: none;
  opacity: 1;
}
.ssrAnimation {
  animation-delay: 2s;
  animation-duration: 0.3s;
  animation-name: toggle;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
}
@keyframes toggle {
  from {
    visibility: hidden;
    opacity: 0;
  }
  to {
    visibility: visible;
    opacity: 1;
  }
}
