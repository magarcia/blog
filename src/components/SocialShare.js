import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Twitter from './icons/twitter';
import Facebook from './icons/facebook';
import LinkedIn from './icons/linkedin';
import './SocialShare.css';

const NETWORKS = [
  {
    name: 'Twitter',
    urlBuilder: (title, url) =>
      `https://twitter.com/intent/tweet?text=${title}&url=${url}&via=martinprins`,
    logo: <Twitter width={24} height={24} title="Share on Twitter" />
  },
  {
    name: 'Facebook',
    urlBuilder: (_, url) => `https://www.facebook.com/sharer/sharer.php?u=${url}`,
    logo: <Facebook width={24} height={24} title="Share on Facebook" />
  },
  {
    name: 'LinkedIn',
    urlBuilder: (title, url) =>
      `https://www.linkedin.com/shareArticle?mini=true&url=${url}&title=${title}&source=https%3A//www.linkedin.com/in/martingarciamonterde/`,
    logo: <LinkedIn width={24} height={24} title="Share on LinkedIn" />
  }
];
export default class SocialShare extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  };

  state = {
    visible: false,
    ssr: true
  };

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll.bind(this));
    this.setState({
      visible: window.pageYOffset > 90,
      ssr: false
    });
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll.bind(this));
  }

  handleScroll() {
    this.setState({
      visible: window.pageYOffset > 90
    });
  }

  render() {
    const { title, url } = this.props;
    const { visible, ssr } = this.state;
    return (
      <aside className={`aside ${visible ? 'visible' : ''} ${ssr ? 'ssrAnimation' : ''}`}>
        {NETWORKS.map(({ name, urlBuilder, logo }) => (
          <a key={name} target="_blank" rel="noopener noreferrer" href={urlBuilder(title, url)}>
            {logo}
          </a>
        ))}
      </aside>
    );
  }
}
