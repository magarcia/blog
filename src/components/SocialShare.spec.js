import React from 'react';
import renderer from 'react-test-renderer';
import SocialShare from './SocialShare';

it('renders correctly', () => {
  const props = {
    title: 'title',
    url: 'http://example.com',
  };
  const tree = renderer.create(<SocialShare {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
