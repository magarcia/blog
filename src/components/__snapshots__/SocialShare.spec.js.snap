// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<aside
  className="aside  "
>
  <a
    href="https://twitter.com/intent/tweet?text=title&url=http://example.com&via=martinprins"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      className="feather feather-twitter"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Share on Twitter
      </title>
      <g
        stroke="none"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M22.162 5.656a8.384 8.384 0 0 1-2.402.658A4.196 4.196 0 0 0 21.6 4c-.82.488-1.719.83-2.656 1.015a4.182 4.182 0 0 0-7.126 3.814 11.874 11.874 0 0 1-8.62-4.37 4.168 4.168 0 0 0-.566 2.103c0 1.45.738 2.731 1.86 3.481a4.168 4.168 0 0 1-1.894-.523v.052a4.185 4.185 0 0 0 3.355 4.101 4.21 4.21 0 0 1-1.89.072A4.185 4.185 0 0 0 7.97 16.65a8.394 8.394 0 0 1-6.191 1.732 11.83 11.83 0 0 0 6.41 1.88c7.693 0 11.9-6.373 11.9-11.9 0-.18-.005-.362-.013-.54a8.496 8.496 0 0 0 2.087-2.165z"
        />
      </g>
    </svg>
  </a>
  <a
    href="https://www.facebook.com/sharer/sharer.php?u=http://example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      className="feather feather-facebook"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Share on Facebook
      </title>
      <g
        stroke="none"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.402 21v-6.966h2.333l.349-2.708h-2.682V9.598c0-.784.218-1.319 1.342-1.319h1.434V5.857a19.19 19.19 0 0 0-2.09-.107c-2.067 0-3.482 1.262-3.482 3.58v1.996h-2.338v2.708h2.338V21H4a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h16a1 1 0 0 1 1 1v16a1 1 0 0 1-1 1h-4.598z"
        />
      </g>
    </svg>
  </a>
  <a
    href="https://www.linkedin.com/shareArticle?mini=true&url=http://example.com&title=title&source=https%3A//www.linkedin.com/in/martingarciamonterde/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      className="feather feather-linkedin"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Share on LinkedIn
      </title>
      <g
        stroke="none"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M18.335 18.339H15.67v-4.177c0-.996-.02-2.278-1.39-2.278-1.389 0-1.601 1.084-1.601 2.205v4.25h-2.666V9.75h2.56v1.17h.035c.358-.674 1.228-1.387 2.528-1.387 2.7 0 3.2 1.778 3.2 4.091v4.715zM7.003 8.575a1.546 1.546 0 0 1-1.548-1.549 1.548 1.548 0 1 1 1.547 1.549zm1.336 9.764H5.666V9.75H8.34v8.589zM19.67 3H4.329C3.593 3 3 3.58 3 4.297v15.406C3 20.42 3.594 21 4.328 21h15.338C20.4 21 21 20.42 21 19.703V4.297C21 3.58 20.4 3 19.666 3h.003z"
          fillRule="nonzero"
        />
      </g>
    </svg>
  </a>
</aside>
`;
